### PR TITLE
Changes for precision

### PR DESCRIPTION
**Pull request by sachingrover211**
This was not an error, but found out the optimal value that should be set for the calculations to be correct as well.

### CODE REVIEW

**Comment by sachingrover211** [comment]
Rest of the error messages will be handled with error messages card

File: newton-raphson.js [hunk 1]
```diff
@@ -4,6 +4,9 @@ define(["parser/parser", "./matrix"], function(Parser, Matrix){
 	var variables = [];
 	var epsilon = 1e-6;
 	var stoppingCriterion = 1e-3;
+	var precision = 12;
+	var n = 1;
+	var m = 0;
```
Comment (jonwetzel) [inline newton-raphson.js:9]: @sachingrover211  Why are n and m declared here?  Are they likely to be re-used anywhere else in the program in the future?  If so there should be documentation here.  In the place where they do get used (getInitialPoint), it seems like they are never modified...I think I must be missing something.

**Review by jonwetzel — COMMENTED** [review]
One clarification question.

**Comment by jonwetzel** [comment]
New error message code (commit 2aa9aff) looks good.

File: newton-raphson.js (hunk 1 above)
Comment (sachingrover211) [reply newton-raphson.js:9]: I was trying to experiment with the ranges. So quantities that can effect the result are at the top. That is the only idea here.

Comment (sachingrover211) [reply newton-raphson.js:9]: @jonwetzel let me know what do you think about it after the response.

File: matrix.js [hunk 2]
```diff
@@ -74,7 +83,10 @@ define([], function(){
 		var isValid = _validateMatrices(a);
 
 		if(!isValid.inv)
-			throw new Error("Matrix can not be inverted as it is not a square Matrix");
+			throw {
+				type: "decomposition",
+				message: messages.get("decomposition");
```
Comment (dharanisakary) [inline matrix.js:88]: semicolon should be removed

File: matrix.js [hunk 3]
```diff
@@ -153,7 +165,10 @@ define([], function(){
 			} else {
 				var pivot = _pivot(U, i);
 				if(pivot == i)
-					throw new Error("Singular matrix, LU decomposition not feasible");
+					throw {
+						type: "decomposition",
+						message: messages.get("decomposition");
```
Comment (dharanisakary) [inline matrix.js:170]: semicolon

File: matrix.js [hunk 4]
```diff
@@ -265,8 +289,10 @@ define([], function(){
 				_matrixThreeArgument(arguments[0], arguments[1], arguments[2]);
 				break;
 			default:
-				throw new Error("Wrong initialization of Matrix class");
-				break;
+				throw {
+					type: "matrix.initialization.incorrect",
+					message: messages.get("matrix.initialization.incorrect");
```
Comment (dharanisakary) [inline matrix.js:294]: semicolon

**Comment by dharanisakary** [comment]
When I try to solve x=y and 2x+y=6, I get an error "Uncaught Error: parse error [column 4]: unexpected variable" even if there is solution(2,2) to this.

**Comment by dharanisakary** [comment]
When I give an inconsistent set of equations, say, x=y and x=y+3, the system gives null, null and after few tries gives an Error in console that "Uncaught Error: Singular Matrix provided".

File: main.js [hunk 5]
```diff
@@ -123,7 +127,7 @@ define([
 				s = new Solver(_equations);
```
Comment (dharanisakary) [inline main.js:127]: Should this be solver defined in line 125?

Comment (sachingrover211) [reply main.js:127]: I had to put it in try catch block and access it outside it as well. #JavaHabits die hard!

File: matrix.js (hunk 3 above)
Comment (sachingrover211) [reply matrix.js:170]: good catch. I fixed it in pull request number 5. Sorry it was left here. I will fix it here as well.

**Comment by sachingrover211** [comment]
The second is expected behavior. Those are singular equations as in there is no solution for that.

**Comment by sachingrover211** [comment]
For first one -  "x=y" "2*x+y=6" equation has to be entered like this in new line. Parser assumes 2x to be a variable and then throws an error saying that variable can not start with a number.

**Comment by sachingrover211** [comment]
@dharanisakary would it be possible to do the testing on error_framework branch. That is the more latest branch and all the error messages have been fixed there. I committed something in this which I wanted to keep it separate. And I just want this merged to master first and then do it for error_framework.

**Pull request merged**